haystack/nodes/answer_generator/openai.py

```diff
@@ -204,59 +259,73 @@ def predict(
         result = {"query": query, "answers": answers}
         return result
 
+    @staticmethod
+    def _create_context(documents: List[Document], join_str: str = " ") -> str:
+        """Join the documents together to create a single context to be used in the PromptTemplate."""
+        doc_contents = [doc.content for doc in documents]
+        context = join_str.join(reversed(doc_contents))
+        return context
+
     def _build_prompt(self, query: str, documents: List[Document]) -> Tuple[str, List[Document]]:
         """
         Builds the prompt for the GPT-3 model so that it can generate an Answer.
         """
-        example_context = f"===\nContext: {self.examples_context}\n===\n"
-        example_prompts = "\n---\n".join([f"Q: {question}\nA: {answer}" for question, answer in self.examples])
-        instruction = "Please answer the question according to the above context.\n" + example_context + example_prompts
-        instruction = f"{instruction.strip()}\n\n"
-
+        example_prompts = "\n---\n".join([f"Q: {query}\nA: {answer}" for query, answer in self.examples])
```

Comment:
This seems a bit hacky to me and limits the options of users to fully control the created prompt (they are now always putting in the examples exactly this way here). However, I see why this is needed at the moment as the PromptTemplate doesn't have a good mechanism to inject examples. This should be addressed on the PromptTemplate side and the refactored here in a later PR.

Reply:
Yes I agree this is hacky, and as you pointed out I think it is necessary at the moment since PromptTemplate can't easily handle lists of items. Similarly, joining the documents together to create the context is now only somewhat controlled by passing the `context_join_str`. 

I think with the introduction of the `prepare` and `post_process` to the `PrompTemplate` in this [PR](https://github.com/deepset-ai/haystack/pull/4061) we would be able to create a new PromptTemplate class (that subclasses from PromptTemplate) that could handle this more elegantly I think.

Reply:
Hmm actually, @tholor would it make more sense to replace 
```python
example_prompts = "\n---\n".join([f"Q: {query}\nA: {answer}" for query, answer in self.examples])
```
with 
```
example_prompts = "\n---\n".join([f"{query}\n{answer}" for query, answer in self.examples])
```
and update the default for `examples` to 
```python
examples = [["Q: What is human life expectancy in the United States?", "A: 78 years."]]
```
This would remove the need for some of the hackyness and allow users to specify how the examples structure should look.

Reply:
yes, good idea. That's better :)
We should still refactor once we have the example logic in the PromptTemplate.